### PR TITLE
docs: refactor to use Furo

### PR DESCRIPTION
**Pull request by miketheman**
Modernize docs to use a cleaner, responsive design.

See https://pradyunsg.me/furo/ for more.

Signed-off-by: Mike Fiedler <miketheman@gmail.com>

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [n/a] Testing strategy is described if automated tests are not included in the PR.
- [n/a] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [n/a] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [n/a] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Author is aware of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer is aware of, and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

**Comment by miketheman** [comment]
This refactor deserves some more scrutiny. It's more modern, response, includes sidebars on left **and** right, and even supports Dark Mode.

**Review by Kyle-Verhoog — APPROVED** [review]
wow this is awesome! Got the official [brand colour](https://www.datadoghq.com/about/resources/) and all, thank you!

I validated the changes using: https://output.circle-artifacts.com/output/job/d285210f-d496-4f3d-ae89-8461c70eb777/artifacts/0/tmp/docs/index.html

The right hand nav position is so great to have on our longer pages:
image

File: docs/conf.py [hunk 1]
```diff
@@ -169,17 +170,25 @@ def _skip(self, word):
...
-    "page_width": "1200px",
-    "sidebar_width": "400px",
+    "light_css_variables": {
+        "color-brand-primary": DATADOG_DARK_PURPLE,
+        "color-brand-content": DATADOG_DARK_PURPLE,
+    },
+    "dark_css_variables": {
+        # TODO: update to use a correct color, since dark purple is hard to read
+        # "color-brand-primary": DATADOG_DARK_PURPLE,
+        # "color-brand-content": DATADOG_DARK_PURPLE,
```
Comment (brettlangdon) [inline docs/conf.py:190]: `rgba(255, 255, 255, 0.76)` is the text color we use on app.datadoghq.com when in dark mode.

?

**Review by brettlangdon — COMMENTED** [review]
one suggestion, but otherwise lgtm

File: docs/conf.py (hunk 1 above)
Comment (miketheman) [reply docs/conf.py:190]: I tried with the hex-equivalent of `#FFFFFFC2` and it looks not-great, as the text is already `#ffffffcc` and the links are semi-indistinguishable:

Screenshot 2023-02-08 at 5 53 50 PM


versus current:

Screenshot 2023-02-08 at 5 54 04 PM

Comment (brettlangdon) [reply docs/conf.py:190]: 👍🏻 makes sense. happy to defer to a later date.

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**